### ISSUE TITLE
Suggestions

### ISSUE THREAD

**Issue by marcosvkp** [JackNoordhuis/Battles#1]
Ideas:

- Add StartBattleEvent ( condition: preStart, posStarted )
- Add EndBattleEvent ( condition: onDeath, timeout, server closed )

- More Modes Customizable, 1v1, 2v2, 3v3
- Customize PlayerSession.
- Load lib( scripts ) extended by abstract script class

**Comment by JackNoordhuis** [comment]
Those events are planned and there will be support for any sized team battle out of the box.

Customising the player session and modifying the battles should be done through other plugins. I'll be providing some example plugins that extend and add onto the core plugins functionality once everything is somewhat working and completed.

**Comment by marcosvkp** [comment]
I have several ideas that would make the plugin something very versatile, for thousands of ways. and about a custom API for, PlayerSession, is it possible to do?

**Comment by JackNoordhuis** [comment]
"Custom API" I plan to do something similar to what I'm doing in the BattleCreationEvent so you can set the class to a custom class through a plugin. I may implement a system to register some sort of extra data handlers without modifying the player session class itself but I don't see any use cases for it at the moment as you should already be able to achieve everything you want without them.

**Comment by marcosvkp** [comment]
I think something like LibLoader of external scripts is possible, extending an abstract class, so I can get the data

**Comment by JackNoordhuis** [comment]
I'm not going to add a script loader, the plugin system in PocketMine is capable of handling whatever you want to achieve.

**Comment by marcosvkp** [comment]
yes it is possible, but I think extending would be better, because there would be simpler and would not have to create a plugin just for this

**Comment by JackNoordhuis** [comment]
Then write your own plugin to load scripts that allow you to modify a battle once the API is in place. The point of having a flexible and extendable codebase/API is so that people can achieve their own requirements without building every use case into the core codebase.

**Comment by daniel022302** [comment]
Commands to start battle?
